projects/plugins/jetpack/_inc/client/components/jetpack-notices/user-license-activation.jsx

```diff
@@ -0,0 +1,111 @@
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+import React, { createElement, useCallback } from 'react';
+import { connect } from 'react-redux';
+import { createInterpolateElement } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import analytics from 'lib/analytics';
+import { getSiteAdminUrl } from 'state/initial-state';
+import {
+	getDetachedLicensesCount,
+	getActivationNoticeDismissInfo,
+	updateLicensingActivationNoticeDismiss as updateLicensingActivationNoticeDismissAction,
+} from 'state/licensing';
+import SimpleNotice from 'components/notice';
+
+/**
+ * Jetpack "user"-licenses activation notice. (a license key is available for activation)
+ *
+ * @param {object} props - The properties.
+ * @param {number} props.detachedLicensesCount - The user's number of "detached" licenses.
+ * @param {object} props.activationNoticeDismissInfo - Object containing `last_detached_count` and `last_dismissed_time`.
+ * @param {Function} props.updateLicensingActivationNoticeDismiss - Function to update the notification dismiss info.
+ * @returns {React.Component} The `UserLicenseActivationNotice` component.
+ */
+const UserLicenseActivationNotice = props => {
+	const DAY_IN_MILLISECONDS = 24 * 3600 * 1000;
+	const MAX_DAYS_DISMISSED = 14;
+
+	const {
+		detachedLicensesCount,
+		activationNoticeDismissInfo,
+		updateLicensingActivationNoticeDismiss,
+		siteAdminUrl,
+	} = props;
+
+	const {
+		last_detached_count: lastDetachedCount,
+		last_dismissed_time: lastDismissedDateTime,
+	} = activationNoticeDismissInfo;
+
+	// TODO: Update this link to point to the user-license activation route.
+	const USER_LICENSE_ACTIVATION_ROUTE = `${ siteAdminUrl }admin.php?page=jetpack#/my-plan`;
+
+	const userHasDetachedLicenses = !! detachedLicensesCount;
+	const userHasNewDetachedLicenses = detachedLicensesCount > ( lastDetachedCount || 0 );
+
+	const now = new Date();
+	const lastDismissedTime = new Date( lastDismissedDateTime ? lastDismissedDateTime : new Date() );
+	const daysNoticeHasBeenDismissed = ( now - lastDismissedTime ) / DAY_IN_MILLISECONDS;
+
+	const trackClick = useCallback( () => {
+		analytics.tracks.recordJetpackClick( 'activate-user-license-notice-click' );
```

Comment:
Should we track the dismissals as well?

Also, super nitpicks:

I'm wondering if we need to add "click" here, since this is already tracked as a `jetpack_wpa_click` action, and if we should use underscores instead of hyphens? We seem to be using underscores in other places:
https://github.com/Automattic/jetpack/blob/39fa657bfe0aaca179275331752fd99a74ac1c41/projects/plugins/jetpack/_inc/client/at-a-glance/stats/dash-stats-bottom.jsx#L55-L57

Reply:
> Should we track the dismissals as well?
Yes definitely, good catch, thank you. :)  I added tracks events for the notice 'view', 'click', and 'dismiss' events.